### ISSUE TITLE
writing to hdf

### ISSUE THREAD

**Issue by HenryGeorgist** [water-tech-repos/wat-hydrograph-stats-py#20]
did @slawler mention that we needed to support writing to the ras tmp hdf file? I dont see that work in here yet.

**Comment by thwllms** [comment]
@HenryGeorgist yeah, he and I talked about that last week. Actively working on it with @barne856 (ARC / Michael Baker). After thinking about it for a bit I ultimately made a new repo.

https://github.com/water-tech-repos/wat-ras-remodeler

Given an existing RAS plan HDF, create a new "temp" HDF (i.e. without a "Results" section)
```
$ ./ras_remodeler.py create-plan-tmp-hdf Muncie.p04.hdf Muncie.p04.tmp.hdf
```

Set a hydrograph in a RAS plan HDF file:
```
$ ./ras_remodeler.py set-hdf-hydrograph Muncie.p04.tmp.hdf "River: White  Reach: Muncie  RS: 15696.24" ./hms-model-output.dss://internal/dss/path/to/specific/hydrograph
```

What do you think? Thought that since this DSS -> HDF aspect is about manipulating RAS data it might be better separate from the initial generic hydrograph "stats" stuff in `wat-hydrograph-stats-py`. Could collect other tools to rework RAS data in `wat-ras-remodeler` over time as needed.

**Comment by HenryGeorgist** [comment]
@thwllms in general that is great.
we do plan a change to the bxx file to update breach elevations so the idea of other tools is good. we have been saying "ras-mutator" but remodeler is fine too.

**Comment by thwllms** [comment]
For the `.b**` breach files, are you referring to this value here (or `Initial Piping Elev` or `Starting WS`)? Are there other breach parameters you are looking at for WAT runs?

![image](https://user-images.githubusercontent.com/5504267/192621583-eca3fc27-d740-467a-8dd9-80d5158e7601.png)
![image](https://user-images.githubusercontent.com/5504267/192621898-59ad22bc-1083-4338-82f9-78f8ad7697e7.png)

**Comment by HenryGeorgist** [comment]
no actually we need the starting wse 947.

Ultimately basically every parameter in that table will be something we want to modify

**Comment by HenryGeorgist** [comment]
reference project  https://github.com/USACE/fragility-curves

**Comment by HenryGeorgist** [comment]
```json
{"results":[{"location":"Muncie System South","failure_elevation":949.7188607392218}]}
```

i think we should make the location name match the lateral structure name.

**Comment by HenryGeorgist** [comment]
here is the sample fragility curve that values are pulled from during a compute 

```json
{
    "name":"Muncie System 3905270004",
	"locations": [{
		"location": "Muncie System South",
		"stage-probability": {
			"xvalues": [0.0, 1.70985E-05, 0.00010452, 0.000564459, 0.002612489, 0.004767421, 0.012118654, 0.037905074, 1.0],
			"ydistributions": [{
				"type": "DeterministicDistribution",
				"parameters": {
					"value": 941.999999
				}
			}, {
				"type": "DeterministicDistribution",
				"parameters": {
					"value": 942.0
				}
			}, {
				"type": "DeterministicDistribution",
				"parameters": {
					"value": 942.9365794
				}
			}, {
				"type": "DeterministicDistribution",
				"parameters": {
					"value": 944.2074385
				}
			}, {
				"type": "DeterministicDistribution",
				"parameters": {
					"value": 946.2217024
				}
			}, {
				"type": "DeterministicDistribution",
				"parameters": {
					"value": 947.1582819
				}
			}, {
				"type": "DeterministicDistribution",
				"parameters": {
					"value": 948.4291409
				}
			}, {
				"type": "DeterministicDistribution",
				"parameters": {
					"value": 949.7
				}
			}, {
				"type": "DeterministicDistribution",
				"parameters": {
					"value": 949.72
				}
			}]
		}
	}]
}
```

**Comment by HenryGeorgist** [comment]
based on that fragility curve, a value of 947 will occur less than 3% of the time

**Comment by thwllms** [comment]
@HenryGeorgist so something like this might be useful?
```
$ ./ras_remodeler.py set-levee-breach Muncie.b04 "White Muncie 13214" --starting-ws 949.7188607392218 --initial-piping-elev 943.21 --center-station 654.32
```

Realizing though that the number of parameters potentially involved could make CLI arguments too unwieldy.

**Comment by HenryGeorgist** [comment]
I think a json output spec that we respect is more ideal.

Thanks
Will

Sent with BlackBerry Work
(www.blackberry.com)


From: Thomas Williams ***@***.******@***.***>>
Date: Tuesday, Sep 27, 2022, 3:23 PM
To: water-tech-repos/wat-hydrograph-stats-py ***@***.******@***.***>>
Cc: Lehman, William P CIV USARMY CEIWR (USA) ***@***.******@***.***>>, Mention ***@***.******@***.***>>
Subject: [URL Verdict: Neutral][Non-DoD Source] Re: [water-tech-repos/wat-hydrograph-stats-py] writing to hdf (Issue #20)


@HenryGeorgist so something like this might be useful?

$ ./ras_remodeler.py set-levee-breach Muncie.b04 "White Muncie 13214" --starting-ws 949.7188607392218 --initial-piping-elev 943.21 --center-station 654.32


Realizing though that the number of parameters potentially involved could make CLI arguments too unwieldy.
